Libraries/DirectMLX.h

```diff
@@ -82,7 +86,7 @@ inline UINT64 DMLCalcBufferTensorSize(
     UINT64 minimumImpliedSizeInBytes = 0;
     if (!strides)
     {
-        minimumImpliedSizeInBytes = sizes[0];
+        minimumImpliedSizeInBytes = dimensionCount > 0 ? sizes[0] : 1;
```

Comment:
I'm curious, what prompted this change? DML doesn't have the notion of a tensor with a `dimensionCount` of 0. If this is for handling scalars, this is probably best handled at a higher level. For example, in TensorFlow we usually set `{1}` for the sizes of scalars before passing it to dmlx.

Reply:
I am doing this to handle scalars. If you think it'll be best handled at a higher level, I can see if I can do that.

Reply:
Appears DMLX already handles things correctly in the general case when strides are passed, and it would be goodness for consistency that DMLX handle the strides=nullptr case properly too. Adding scalars in API validation is something Adrian and I have discussed a few times before, but just not added yet. So having DMLX.h return a sane answer in the general case is nice.

```
        minimumImpliedSizeInBytes = 1;
        for (UINT i = 0; i < dimensionCount; ++i)
        {
            minimumImpliedSizeInBytes *= sizes[i];
        }
        minimumImpliedSizeInBytes *= elementSizeInBytes;
```